### PR TITLE
optimize CodeGenF::on_finish() performed in a single thread

### PR DESCRIPTION
**Pull request by unserialize**
# This PR speeds up "CodeGen on_finish" single-thread step

image

Dropped, or paralleled, or removed some computations that were previously done in a single thread:
* Don't get a vector of functions and sort it
* Extract preparing functions to execute() to be run in parallel.
* Async launch of tl-schema generation and other previously sync tasks.

### CODE REVIEW

File: compiler/pipes/code-gen.cpp [hunk 1]
```diff
@@ -154,34 +132,12 @@ void CodeGenF::prepare_generate_function(FunctionPtr func) {
     ? func->file_id->owner_lib->headers_dir() + func->header_name
     : func->subdir + "/" + func->header_name;
 
-  my_unique(&func->static_var_ids);
-  my_unique(&func->global_var_ids);
-  my_unique(&func->local_var_ids);
+  std::sort(func->static_var_ids.begin(), func->static_var_ids.end());
```
Comment (AlexK0) [inline compiler/pipes/code-gen.cpp:135]: Why did you remove unique?

Comment (unserialize) [reply compiler/pipes/code-gen.cpp:135]: Because these vars are already unique. I mean, that kphp_assert(count_before == count_after) is true always.

File: compiler/pipes/code-gen.h [hunk 2]
```diff
@@ -14,15 +14,14 @@
 class CodeGenerator;
 
 class CodeGenF final : public SyncPipeF<FunctionPtr, WriterData> {
-  using need_profiler = std::false_type;
+  using need_profiler = std::true_type;
```
Comment (DrDet) [inline compiler/pipes/code-gen.h:17]: Why was it disabled before?

Comment (unserialize) [reply compiler/pipes/code-gen.h:17]: Because it had no logic for each function, it had only `on_finish`. That's why, it was disabled, as it always showed zero measurements.
Now, I've implemented `CodeGenF::execute` that works in parallel.

**Review by DrDet — APPROVED** [review]
